group4/81960_Victoria_Angelova/practicum_week_5/programmer.cpp

```diff
@@ -0,0 +1,82 @@
+#include "programmer.h"
+
+void Programmer::copy(const std::string &_name, const std::string &_spec,
+                      const double &_hours, const std::string &_level)
+{
+    name = _name;
+    specialization = _spec;
+    hoursWorked = _hours;
+    level = _level;
+}
+
+void Programmer::decreaseHours(const double &_hours)
+{
+    hoursWorked -= _hours;
+}
+
+Programmer::Programmer(const std::string &_name, const std::string &_spec,
+                       const double &_hours, const std::string &_level)
+{
+    copy(_name, _spec, _hours, _level);
+}
+
+std::string Programmer::getName() const
+{
+    return name;
+}
+
+std::string Programmer::getSpecialization() const
+{
+    return specialization;
+}
+
+std::string Programmer::getLevel() const
+{
+    return level;
+}
+double Programmer::getHoursWorked() const
+{
+    return hoursWorked;
+}
+
+bool Programmer::canBeAssigned(Task *newTask) const
+{
+    if (getSpecialization() != newTask->getTaskType())
+    {
+        return false;
+    }
+    if (getLevel() == "junior" && newTask->getExecutionTime() * 4 <= getHoursWorked())

```

Comment:
This and next 2 are almost the same but difficult to read. It's better to calculate the needed hours in advance based on the level and then make one comparison.